compiler/modules/types/core.bal

```diff
@@ -97,6 +97,10 @@ public isolated class Env {
         }
     }
 
+    isolated function cellAtomType(Atom atom) returns CellAtomicType {
```

Comment:
This shouldn't be a member function of Env, because it doesn't use the state in Env.  If you make it a module-level function, then I think there will be several other functions that do not need a Context or Env argument.